### PR TITLE
DEV-5000: secure-messenger  audit and readme

### PR DESCRIPTION
**Pull request by MSilva95**
fix audit and readme for  secure-messenger

### CODE REVIEW

File: subjects/mobile-dev/secure-messenger/README.md [hunk 1]
```diff
@@ -1,36 +1,30 @@
...
 ### Bonus:
 
-Group messaging: Users should be able to create or join group chats with multiple users having at least the same functionality as regular chats.
+- **Group messaging**, where users are able to create or join group chats with multiple users having at least the same functionality as regular chats.
 
-Push notifications: Users should receive notifications of new messages or friend requests using the firebase_messaging package.
+- **Push notifications**, where users can receive notifications of new messages or friend requests using the firebase_messaging package.
 
-## Suggested Packages for Flutter and Firebase:
+#### Suggested Packages for Flutter and Firebase:
```
Comment (nprimo) [inline subjects/mobile-dev/secure-messenger/README.md:27]: I would include the link to the official documentation https://firebase.google.com/docs/auth/flutter/start

File: subjects/mobile-dev/secure-messenger/README.md [hunk 2]
```diff
@@ -1,36 +1,30 @@
...
+  - **edit** or **delete** messages.
+- **Secret chats**: Users must be able to start encrypted one-on-one chats with the same functionality as regular chats with one exception: all messages have to be encrypted using an end-to-end algorithm.
 
 ### Bonus:
 
-Group messaging: Users should be able to create or join group chats with multiple users having at least the same functionality as regular chats.
+- **Group messaging**, where users are able to create or join group chats with multiple users having at least the same functionality as regular chats.
 
-Push notifications: Users should receive notifications of new messages or friend requests using the firebase_messaging package.
+- **Push notifications**, where users can receive notifications of new messages or friend requests using the firebase_messaging package.
```
Comment (nprimo) [inline subjects/mobile-dev/secure-messenger/README.md:25]: Nothing major, but I would use some formatting to highlight `firebase_messaging`

File: subjects/mobile-dev/secure-messenger/README.md [hunk 3]
```diff
@@ -1,36 +1,30 @@
...
-- Search users: Users should be able to search for other users by username or other information.
-- Add contacts: Users should be able to add contacts by searching for their username or scanning a QR code.
-- Messaging: Users should be able to send text messages, images, and videos, with features such as read receipts, typing indicators, and the ability to edit or delete messages.
-- Secret chats: Users should be able to start encrypted one-on-one chats with the same functionality as regular chats with one exception: all messages have to be encrypted using an end-to-end algorithm.
+- **Login/Signup**: Implement a login/signup page and the necessary functionality for user authentication and account creation.
+- **User profile**: Users must be able to create a profile with a profile picture, username, and other information.
+- **Search users**: Users must be able to search for other users by username or other information.
+- **Add contacts**: Users must be able to add contacts by searching for their username or scanning a QR code.
+- **Messaging**: Users must be able to send text messages, images, and videos. Apart from that you must implement these features:
+  - **read receipts**
```
Comment (nprimo) [inline subjects/mobile-dev/secure-messenger/README.md:16]: I would clarify better this feature - I only understood what was required after reading the audit. I would rephrase this or add more information like `the app should display when the message has been read`

File: subjects/mobile-dev/secure-messenger/audit/README.md [hunk 4]
```diff
@@ -1,23 +1,69 @@
...
+
+##### Check if the app include a secret chat feature where messages are encrypted end-to-end, meaning that the messages must only be visible to users, but not on the backend of the app.
+
+###### Can users start encrypted one-on-one chats in secret chats?
+
+###### Are all messages in secret chats encrypted using an end-to-end algorithm? (Ask the student to show you if necessary)
+
+###### Does the app support both iOS and Android platforms?
+
+#### Bonus
```
Comment (nprimo) [inline subjects/mobile-dev/secure-messenger/audit/README.md:61]: These bonus questions are not in line with the subject. In the subject, as a bonus, it is asked to add:
- group chats
- push notifications

File: subjects/mobile-dev/secure-messenger/README.md [hunk 5]
```diff
@@ -1,36 +1,30 @@
...
-Implement end-to-end encryption for secret chats using the encrypt or webcrypto packages. Use a symmetric encryption algorithm like RSA, and generate a new public and private key pair for each pair of users for each secret chat. The public keys should be securely shared between the sender and receiver.
-
-### Implement the following features:
-
-- User profiles: Users should be able to create a profile with a profile picture, username, and other information.
-- Search users: Users should be able to search for other users by username or other information.
-- Add contacts: Users should be able to add contacts by searching for their username or scanning a QR code.
-- Messaging: Users should be able to send text messages, images, and videos, with features such as read receipts, typing indicators, and the ability to edit or delete messages.
-- Secret chats: Users should be able to start encrypted one-on-one chats with the same functionality as regular chats with one exception: all messages have to be encrypted using an end-to-end algorithm.
+- **Login/Signup**: Implement a login/signup page and the necessary functionality for user authentication and account creation.
```
[the remark this answers is not in the document]
Comment (MSilva95) [reply subjects/mobile-dev/secure-messenger/README.md]: done

File: subjects/mobile-dev/secure-messenger/README.md (hunk 2 above)
Comment (MSilva95) [reply subjects/mobile-dev/secure-messenger/README.md]: done

File: subjects/mobile-dev/secure-messenger/README.md (hunk 1 above)
Comment (MSilva95) [reply subjects/mobile-dev/secure-messenger/README.md:35]: added links to all

**Review by nprimo — APPROVED** [review]
I left a small comment about some of the bonus questions of the audit - I leave it to you if you want to take some action or not about it :)

File: subjects/mobile-dev/secure-messenger/audit/README.md [hunk 6]
```diff
@@ -1,23 +1,77 @@
...
+
+###### +Can users create or join group chats with multiple users?
+
+###### +Do group chats have at least the same functionality as regular chats?
+
+###### +Do users receive push notifications when they receive new messages?
+
+###### +Are users able to receive push notifications for friend requests?
+
+###### +Does the app provide appropriate error handling and error messages?
```
[the remark this answers is not in the document]
Comment (MSilva95) [reply subjects/mobile-dev/secure-messenger/audit/README.md:71]: done
